.github/workflows/test.yml

```diff
@@ -0,0 +1,31 @@
+name: Test
+# Make sure only one action triggers the job, otherwise pushing to a
+# pull-request will run it twice.
+on:
+  pull_request:
+    branches:
+      - "*"
+  merge_group:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version: [3.9, 3.12]
+
+    steps:
+      - name: "Clone Repository"
+        uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          # XXX: install from pyproject.toml
```

Comment:
Do you want to address that in this PR?

Reply:
Sorry, yes! This is hopefully fixed now in the right way.

Reply:
It is :)